src/hashes/ternary/curl_p/batched/bct_curlp.rs

```diff
@@ -0,0 +1,125 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::hashes::ternary::{
+    curl_p::batched::{
+        bct::{BcTrit, BcTritArr, BcTrits},
+        HIGH_BITS, NUM_ROUNDS,
+    },
+    HASH_LENGTH,
+};
+
+pub(crate) struct BctCurlP {
+    state: BcTritArr<{ 3 * HASH_LENGTH }>,
+    scratch_pad: BcTritArr<{ 3 * HASH_LENGTH }>,
+}
+
+impl BctCurlP {
+    #[allow(clippy::assertions_on_constants)]
```

Comment:
There is only one `assert!` that this applies to, so we could move the `#[allow(...)]` directly above it.

Reply:
Actually you can't put attributes on macros